src/mailing/sender.js

```diff
@@ -9,26 +11,28 @@ const transport = nodemailer.createTransport({
   },
 });
 
-// Function to send a test email (promisified)
-async function sendTestEmail(recipient) {
-  return new Promise((resolve, reject) => {
-    transport.sendMail(
-      {
-        from: process.env.EMAIL_USERNAME,
-        to: recipient,
-        subject: 'Test Email',
-        text: 'Hello, this is a test email!',
-        html: '<p>Hello, this is a test email!</p>',
-      },
-      (err, res) => {
-        if (err) {
-          reject(err);
-        } else {
-          resolve(res);
-        }
-      }
-    );
+const baseEmail = new Email({
+  message: {
+    from: 'codethedream.practicum.team4@outlook.com',
+  },
+  transport: transport, // Use the transport you created
+  subjectPrefix: process.env.NODE_ENV === 'production' ? '' : '(TEST) ',
+  views: {
+    root: path.resolve('../templates/test'), // this tells email-templates to look for templates starting from this path
```

Comment:
Your `templates` file is not inside `src`.

You should move the `templates` folder to inside `src` and better yet inside `src/mailing` since those templates will be only used by our email functions so it makes sense for them to live in the same place. 

---

When using `path.resolve` it will calculate the path relative to where node is running from (so usually that will mean relative to the root of the repo). So `../templates/test` is saying "go up one folder ('`..`') from where this backend repo is stored". You can double check this by adding a log somewhere in this file `console.log(path.resolve("../templates/test"));`. On my computer this logs: `/Users/aasante/dev/practicum2023/templates/test` which is _outside_ of my repo folder.

You can compare by also console.logging the Current Working Directory with `console.log(process.cwd());`. On my local computer that prints the directory that my backend repo is at: `/Users/aasante/dev/practicum2023/backend`. 

---

Here we are trying to tell Email-Templates the _root_ folder to look for all templates; not just the specific folder for a single template. So we should be directing it to the `/templates` folder not to the `/templates/test` folder. Because later on as we add more emails, there will be `/templates/forgot_password`, `/templates/order_email` etc. But all of those will use the same view root. Then in the actual email sending function (eg: sendTestEmail) we tell Email-Templates which folder to use via the `template` option.

---

Then, if your templates live at: `<backend_repo_root>/src/mailing/templates` then you can refer to it using any of the following:

*   `path.resolve('src/mailing/templates')` \<-- giving path.resolve the path from the root of the repo / the current working firectory where node is running from
*   `path.resolve(__dirname, "templates")` or `path.resolve(__dirname, "./templates")` \<-- using the special variable `__dirname` which returns the directory of _the file where the code is_ and then telling path to check the same folder/directory and navigate to `templates` folder from there